atc/creds/vault/manager.go

```diff
@@ -89,6 +92,10 @@ func (manager *VaultManager) MarshalJSON() ([]byte, error) {
 func (manager *VaultManager) Config(config map[string]interface{}) error {
 	// apply defaults
 	manager.PathPrefix = "/concourse"
+	manager.LookupTemplates = []string{
```

Comment:
I think we might need to do something different here. I noticed in testing that when configuring as a var source, the defaults don't seem to get replaced - the configuration is additive. They're replaced properly if I configure `CONCOURSE_VAULT_LOOKUP_TEMPLATES` though.

If you need a quick way to try this out, here's my pipeline:

```yaml
var_sources:
- name: myvault
  type: vault
  config:
    url: https://vault:8200
    ca_cert: ((vault_ca_cert))
    client_cert: ((vault_client_cert))
    client_key: ((vault_client_key))
    lookup_templates: [/woahnelly]
    auth_backend: cert

jobs:
- name: use-creds
  plan:
  - task: print-secret
    config:
      platform: linux
      image_resource:
        type: mock
        source: {mirror_self: true}
      run:
        path: /bin/sh
        args:
          - -c
          - |
              echo ((myvault:some-secret))
```

Here's what I did:

```sh
# in concourse repo
$ ./hack/dc up
$ ./hack/vault/init
$ vault secrets enable -version=1 -path=concourse kv
$ vault write concourse/main/vault-repro/some-secret value=hello

# (replace these paths)
$ cat <<EOF > vault-values.yml
vault_ca_cert: $(cat ~/src/concourse/hack/vault/certs/vault-ca.crt | jq -Rs .)
vault_client_cert: $(cat ~/src/concourse/hack/vault/certs/concourse.crt | jq -Rs .)
vault_client_key: $(cat ~/src/concourse/hack/vault/certs/concourse.key | jq -Rs .)
EOF
$ fly -t l sp -p vault-repro -c vault-repro.yml -l vault-values.yml
```

When I triggered `use-creds`, I expected it to only look for `/concourse/woahnelly/some-secret` and result in an error, however it actually finds the credential in the default path and succeeds.

Reply:
Thanks, trying to reproduce this.

Reply:
Ah, got it. I didn't catch this earlier because it happens just when configuring <2 custom templates. They're individually overwriting the entries in the default array, as opposed to replacing it as one unit, or even appending. Looks like behavior from the `mapstructure` library which should be fine to tweak or work around.